src/main/java/htsjdk/samtools/SamInputResource.java

```diff
@@ -89,7 +91,9 @@ public String toString() {
     public static SamInputResource of(final File file) { return new SamInputResource(new FileInputResource(file)); }
 
     /** Creates a {@link SamInputResource} reading from the provided resource, with no index. */
-    public static SamInputResource of(final Path path) { return new SamInputResource(new PathInputResource(path)); }
```

Comment:
Could you add javadoc explaining what the wrapper is for on all these constructors that take one?